library/Notifications/Api/V1/ApiV1.php

```diff
@@ -0,0 +1,333 @@
+            $column = $condition->getColumn();
+            if (! in_array($column, $allowedColumns)) {
+                throw HttpBadRequestException::create([
+                    sprintf(
+                        'Invalid filter column %s given, only %s are allowed',
+                        $column,
+                        preg_replace('/,([^,]*)$/', ' and$1', implode(', ', $allowedColumns))
+                    )
+                ]);
+            }
+
+            if ($column === 'id') {
+                if (! Uuid::isValid($condition->getValue())) {
+                    throw HttpBadRequestException::create(['The given filter id is not a valid UUID']);
+                }
+
+                $condition->setColumn($idColumnName);
+            }
+        };
+    }
+
+    /**
+     * Create an array of response data.
+     *
+     * This method constructs an array containing the status code, body, and headers to create a response.
+     *
+     * @param ?int $statusCode The HTTP status code for the response.
+     * @param ?array $additionalHeaders An associative array of additional headers to include in the response.
+     * @param bool|callable|float|int|Iterator|null|StreamInterface|string $body The body of the response.
+     * @return array An associative array containing 'status', 'body', and 'headers' keys.
+     */
+    protected function createArrayOfResponseData(
+        int $statusCode = null,
+        bool|callable|float|int|Iterator|null|StreamInterface|string $body = null,
+        array $additionalHeaders = null,
+    ): array {
+
+        return array_filter([
+            'status' => $statusCode,
+            'body' => $body,
+            'headers' => $additionalHeaders,
+        ], static fn($v) => $v !== null);
+    }
+
+    /**
+     * Create a ResponseInterface object from the given response data.
+     *
+     * This method constructs a ResponseInterface object using the provided response data array.
+     * It sets the status code, headers, and body of the response based on the data in the array.
+     *
+     * @param array $responseData An associative array containing 'status', 'body', and 'headers' keys.
+     * @return ResponseInterface The constructed ResponseInterface object.
+     */
+    protected function createResponse(array $responseData): ResponseInterface
+    {
+        $response = $this->getResponse();
+
+        if (isset($responseData['status'])) {
+            $response = $response->withStatus($responseData['status']);
+        }
+        if (isset($responseData['headers']) && is_array($responseData['headers'])) {
+            foreach ($responseData['headers'] as $name => $values) {
+                if (is_array($values)) {
+                    foreach ($values as $value) {
+                        $response = $response->withHeader($name, $value);
+                    }
+                } else {
+                    $response = $response->withHeader($name, $values);
+                }
```

Comment:
Each call on `$response` creates a clone. Change it as discussed offline, please.